homeassistant/components/todo/intent.py

```diff
@@ -0,0 +1,78 @@
+"""Intents for the todo integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import intent
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+
+from . import DOMAIN, TodoItem, TodoListEntity
+
+INTENT_ADD_ITEM = "HassListAddItem"
+
+
+async def async_setup_intents(hass: HomeAssistant) -> None:
+    """Set up the todo intents."""
+    intent.async_register(hass, AddItemIntent())
+
+
+class AddItemIntent(intent.IntentHandler):
+    """Handle AddItem intents."""
+
+    intent_type = INTENT_ADD_ITEM
+    slot_schema = {"item": cv.string, vol.Optional("list"): cv.string}
+
+    async def async_handle(self, intent_obj: intent.Intent):
+        """Handle the intent."""
+        hass = intent_obj.hass
+
+        slots = self.async_validate_slots(intent_obj.slots)
+        item = slots["item"]["value"]
+
+        component: EntityComponent[TodoListEntity] = hass.data[DOMAIN]
+        list_entities: list[TodoListEntity] = list(component.entities)
+        if not list_entities:
+            raise intent.IntentHandleError("No list entities")
+
+        target_list: TodoListEntity | None = None
+
+        if "list" in slots:
+            # Add to a list by name
+            list_name = slots["list"]["value"]
+
+            # Try literal name match first
+            for maybe_list in list_entities:
+                if not isinstance(maybe_list.name, str):
```

Comment:
can this happen ?

Reply:
I don't know if it can *actually* happen, but mypy threw a fit until I did this. The definition of entity name includes `UndefinedType`: https://github.com/home-assistant/core/blob/691de148cf29d5636d89dee305428a4ce274e5cd/homeassistant/helpers/entity.py#L470

Reply:
ah check, yeah ok.

Reply:
Using `async_match_states` now